lib/src/page_list_viewport_gestures.dart

```diff
@@ -454,57 +670,85 @@ class DeprecatedPanAndScaleVelocityTracker {
-    // smaller numbers end more quickly.
-    _xSimulation = ClampedSimulation(
-      FrictionSimulation(
-        kHorizontalDrag,
-        _position.dx,
-        _velocity.dx,
-        // Horizontal friction simulation uses the default constantDeceleration.
-        //
-        // constantDeceleration impacts how smoothly the simulation comes to a stop
-        // near the end (when it smoothly slides to zero).
-      ),
-      dxMin: -3000,
-      dxMax: 3000,
-    );
-    _ySimulation = ClampedSimulation(
-      FrictionSimulation(
-        kVerticalDrag,
-        _position.dy,
-        _velocity.dy,
-        // Vertical friction is intentionally different from horizontal friction,
-        // based on empirical results.
-        //
-        // constantDeceleration impacts how smoothly the simulation comes to a stop
-        // near the end (when it smoothly slides to zero).
-        constantDeceleration: 2.35,
-      ),
-      dxMin: -2000, // Scroll down more slowly than scrolling up. A config found in some other scrolling systems.
-      dxMax: 3000,
-    );
+        _velocity = velocity,
+        _momentumSimInitialVelocityScalar = initialVelocityIncreaseScalar,
+        _dragIncreaseScalar = dragIncreaseScalar {
+    if (_velocity.dx.abs() > 0 && _velocity.dy.abs() > 0) {
+      // The simulation is not locked to an axis, it is in an arbitrary direction.
+      _xSimulation = FrictionAndFirstOrderDragMomentumSimulation(
+          staticFrictionCoefficient,
+          horizontalDragCoefficient * _dragIncreaseScalar,
+          mass,
+          _position.dx,
+          _velocity.distance,
+          math.cos(math.atan2(_velocity.dy, _velocity.dx)),
+          initialVelocityScalar: ViewportThresholdsAndScales.diagonalLaunchVelocityIncreaseScalar);
+      _ySimulation = FrictionAndFirstOrderDragMomentumSimulation(
+          staticFrictionCoefficient,
+          horizontalDragCoefficient * _dragIncreaseScalar,
+          mass,
+          _position.dy,
+          _velocity.distance,
+          math.sin(math.atan2(_velocity.dy, _velocity.dx)),
+          initialVelocityScalar: ViewportThresholdsAndScales.diagonalLaunchVelocityIncreaseScalar);
+    } else {
+      // The simulation is locked to one of the axes.
+      _xSimulation = FrictionAndFirstOrderDragMomentumSimulation(
+          staticFrictionCoefficient, verticalDragCoefficient * _dragIncreaseScalar, mass, _position.dx, _velocity.dx, 1,
+          initialVelocityScalar: _momentumSimInitialVelocityScalar);
+      _ySimulation = FrictionAndFirstOrderDragMomentumSimulation(staticFrictionCoefficient,
+          horizontalDragCoefficient * _dragIncreaseScalar, mass, _position.dy, _velocity.dy, 1,
+          initialVelocityScalar: _momentumSimInitialVelocityScalar);
+    }
   }
 
   final Offset _position;
   final Offset _velocity;
-  late final ClampedSimulation _xSimulation;
-  late final ClampedSimulation _ySimulation;
+  final double _momentumSimInitialVelocityScalar;
```

Comment:
These new properties don't seem to be used.

Reply:
I think a couple of lines above that.